### PR TITLE
Install tensorflow with pip (and tensorflow + keras for R)

### PR DESCRIPTION
**Pull request by damianavila**
Let's try with pip first because I am worried using conda will expose us to an endless dependencies resolution calculation.

### CODE REVIEW

**Comment by rabernat** [comment]
In our experience, installing tensorflow reliably into a conda-forge environment, particularly if you need to use the GPU, is very difficult.

Pangeo maintains a docker image with pydata + ML libraries here: https://github.com/pangeo-data/pangeo-docker-images

Perhaps there is something there that can help this effort.

File: environment.yml [hunk 1]
```diff
@@ -55,6 +55,7 @@ dependencies:
     - folium
     - geopy
     - otter-grader==3.1.4
+    - tensorflow
```
Comment (yuvipanda) [inline environment.yml:58]: Let's pin a version here?

I like pip for tensorflow as it's what they recommend doing on their website.

**Comment by sgibson91** [comment]
> I still think the CI needs to be able to succeed in this endeavor to keep things consistent and scalable in the medium/long term.

💯 Testing locally and testing in CI serve different purposes. CI needs to be consistent to reduce toil but we can fall back to local builds for deep debugging with fast turnaround when needed.

**Comment by choldgraf** [comment]
How much of this issue is because we are trying to install tensorflow on top of a bunch of other stuff? Could we:

1. Create a basic "Modern ML environment image" that is similar (or maybe re-uses) the one that @rabernat shared above.
2. Provide some way for people to install packages on top of this image
3. When people want tensorflow or keras etc, we add a new user image to their hub's user_profile list that is designed for ML, rather than lumping it all in one place

?

**Comment by GeorgianaElena** [comment]
> How much of this issue is because we are trying to install tensorflow on top of a bunch of other stuff?

I believe there was also a discussion before (in the context of the huge size of this image) about splitting it into smaller ones, like one for R classes and other for python. A "Modern ML environment image" would make sense too.

**Comment by sgibson91** [comment]
I am +1 on the image splitting idea

**Comment by damianavila** [comment]
Yep, we definitely need some sort of splitting in a hierarchical-based model, and we should aim for that sooner than later, although, most probably, not in this PR, but 💯 to the general sentiment.

Btw, some updates on testing the current image in https://github.com/2i2c-org/utoronto-image/issues/6#issuecomment-1014934639

**Comment by choldgraf** [comment]
I've got an issue to track when/how to recommend multiple user images here: https://github.com/2i2c-org/infrastructure/issues/935

**Review by yuvipanda — APPROVED** [review]

**Comment by yuvipanda** [comment]
I think #12 needs to be merged before this can be, otherwise this will fail to build now.

**Comment by yuvipanda** [comment]
Just gonna merge this, and see what happens.

**Pull request merged**